src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.ActiveDirectory.tests.ps1

```diff
@@ -0,0 +1,61 @@
+Import-Module Microsoft.Graph.Applications
+Import-Module -Name $PSScriptRoot\..\Arcus.Scripting.ActiveDirectory -ErrorAction Stop
+
+InModuleScope Arcus.Scripting.ActiveDirectory {
+    Describe "Arcus Azure Active Directory integration tests" {
+        BeforeEach {
+            $config = & $PSScriptRoot\Load-JsonAppsettings.ps1
+            & $PSScriptRoot\Connect-AzAccountFromConfig.ps1 -config $config
+        }
+        Context "Add an Active Directory Application Role Assignment" {
+            It "Creating a new role and assigning it should succeed" {
+                # Arrange
+                $ClientId = $config.Arcus.ActiveDirectory.MainAppClientId
+                $RoleName = 'SomeRole'
+                $AssignRoleToClientId = $config.Arcus.ActiveDirectory.ClientAppClientId
+
+                # Act
+                {
+                   Add-AzADAppRoleAssignment -ClientId $ClientId -Role $RoleName -AssignRoleToClientId $AssignRoleToClientId
```

Comment:
If we add a role assignment here, should we then remove it again, too, in the teardown of this test? (`try`/`finally`)

Reply:
A bit conflicted on this. The way I have set it up right now is like this:
1. An integration test is executed that creates an approle and assignment.
2. An integration test is executed that gets the approles and assignments from step 1.
3. An integration test is executed that removes the approle and assignments from step 1.

So basically the integration tests itself already act as setting up and cleaning the environment.
However, this might not be the best approach since the integration tests are now sort of "linked" together.

But if I add setup and teardown steps to the integration tests then I would be using the `Arcus.Scripting.ActiveDirectory` scripts in the setup and teardown (since doing it without these scripts I would need quite some extra code) and not sure if that is the way to go, also I would basically just be executing the same logic for each integration test (create, get, remove, where get is optional)...

I tend to lean towards keeping the current setup, as the tests will run successfully even if a remove has not been executed for whatever reason. (adding already existing roles will not cause an issue, getting roles that are not there will not cause an issue, removing roles that are not there will not cause an issue). 
But very interested in your view on this @stijnmoreels 👍🏻

Reply:
Yes, I understand your hesitation on this. The problem I'm seeing though, is if these tests are running successful, even if the result in Active Directory is incorrect, then we would have a lot of false positives, right? How certain are we then if the scripts are running as expected?

Linking tests together is, like you already said, not a very stable approach. If we change the order of the tests, I expect failures, right?

We had the same things in other integration tests, like Key Vault secrets added/removed. It is a good approach to see each integration test as a separate environment and fail on anything that does not match that environment. The teardown, in this case, could also be a lot simpler and I would even make sure that we do not use any Arcus functionality on that, to really make sure that the script we are testing results in the expected results.  

In this case, a call to `New-MgServicePrincipalAppRoleAssignment -ServicePrincipalId $adServicePrincipalRoleAssignTo.Id -PrincipalId $adServicePrincipalRoleAssignTo.Id -ResourceId $adServicePrincipal.Id -AppRoleId $currentAppRole.Id` for example would be enough I think, no? The `PrincipalId`, the `ResourceId`... those are in fact identifiers we could even know beforehand, if that would a bit ease your pain 😄 .

I expect obscure problems if we forget that these tests are linked together. I see the added-value of adding the extra code, but I'm also happy to do it in a follow-up PR if you want to close complete this one. 👍 #teamwork

Reply:
Good points, I made a change for this, see [`ed7f050` (#336)](https://github.com/arcus-azure/arcus.scripting/pull/336/commits/ed7f050be4a534b661f10847c278d94fb1a3b700).
I have used the actual `Arcus.Scripting.ActiveDirectory` like `Add-AzADAppRoleAssignment` to the setup and teardown since otherwise I would need quite some logic (for example, the looping until a role becomes available).

Reply:
Ok, thx, it is already a start to have these ones. 👍